src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> element = new Node<>(tail, value, null);
+            tail.next = element;
+            tail = element;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> nodeByIndex = getNode(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            head = new Node<>(null, value, nodeByIndex);
+            getNodeAndCheckIndex(index).prev = head;
+            size++;
+        } else {
+            Node<T> previous = getNodeAndCheckIndex(index).prev;
+            Node<T> newNode = new Node<>(previous, value, nodeByIndex);
+            previous.next = newNode;
+            nodeByIndex.prev = newNode;
+            size++;
+        }
```

Comment:
we have similar logic here like `size++;` or `getNodeAndCheckIndex(index).prev`
let's move common code from ifs so we will write it only once